lib/Wrapper/OptimizingScheduler.cpp

```diff
@@ -707,6 +786,52 @@ void ScheduleDAGOptSched::finalizeSchedule() {
   });
 }
 
+void ScheduleDAGOptSched::runSchedPass(SchedPassStrategy S) {
+  switch (S) {
+  case OptSchedMinRP:
+    scheduleOptSchedMinRP();
+    break;
+  case OptSchedBalanced:
+    scheduleOptSchedBalanced();
+    break;
+  }
+}
+
+void ScheduleDAGOptSched::scheduleOptSchedMinRP() {
+  LLVM_DEBUG(dbgs() << "Starting first pass through\n");
+  LatencyPrecision = LTP_UNITY;
+  // Set times for the first pass
+  RegionTimeout = FirstPassRegionTimeout;
+  LengthTimeout = FirstPassLengthTimeout;
+  HeurSchedType = SCHED_LIST;
+
+  schedule();
+}
+
+void ScheduleDAGOptSched::scheduleOptSchedBalanced() {
+  LLVM_DEBUG(dbgs() << "Starting second pass through\n");
+  SecondPass = true;
+  LatencyPrecision = LTP_ROUGH;
+  // Set times for the second pass
+  RegionTimeout = SecondPassRegionTimeout;
+  LengthTimeout = SecondPassLengthTimeout;
+  // Set the heuristic for the enumerator in the second pass.
+  EnumPriorities = SecondPassEnumPriorities;
+  // Force the input to the balanced scheduler to be the sequential order of the
+  // (hopefully) good max occupancy schedule. We don’t want the list scheduler
```

Comment:
Occupancy doesn't make sense in a generic context.

Reply:
Fixed